### ISSUE TITLE
installing psycopg2 in lambda

### ISSUE THREAD

**Issue by rae89** [apex/apex#884]
What is the best approach to installing psycopg2 or any other c compiled python library using Apex?

**Comment by rbaudet** [comment]
Hi, are you on Mac or Windows ? 
I guess the problem is that your psycopg2 binary is built for the wrong platform. 
AWS Lambda functions run on Linux, so the binary built with pip for Mac OS and Windows won't run on Linux.
I just discover Apex so I don't have a solution to install it using Apex but I suggest to remove psycopg2 from your requirment.txt and install it manually with Docker. 
You run a Linux container locally and then install psycopg2 with pip in your working directory,  and then you have the Linux version of the library which should work in AWS Lambda.

**Comment by Liath** [comment]
Yeah this sounds an awful lot like you're deploying from something other than linux. Can you put the skeleton of your lambda somewhere so we can reproduce your problem?

**Comment by rae89** [comment]
Yeah, I was deploying from a macos machine.  So it seems like I need to be building the lambda via docker?

**Comment by Liath** [comment]
Dockers sounds like overkill.
Maybe do a something with (environment markers)[https://pip.pypa.io/en/stable/reference/pip_install/#requirement-specifiers]?

That being said, the psycopg docs seem (opposed to using the binary package)[http://initd.org/psycopg/docs/install.html#binary-install-from-pypi]. So Docker may be best.

Sorry I can't be more help, I'm not terribly familiar with python.

**Comment by geraldarthur** [comment]
Lambda builds via Docker worked for me when I ran into this issue last year. Try this in your function.json config
```
"hooks":{
    "build": "docker run --rm -v $(pwd):/var/task lambci/lambda:build-python3.6 pip install --upgrade -r requirements.txt -t ."
},
```
